Beta v6.34.0

MichaIng opened the pull request:
### Beta v6.34.0
_(08/12/20)_

#### Changes / Improvements / Optimisations
- [Raspberry Pi 400](https://dietpi.com/docs/dietpi_sbc/#raspberry-pi) | Correct detection for this new Raspberry Pi model has been added.
- DietPi-Globals | In DietPi scripts, the PATH variable is now overwritten with the Debian/bash system default to assure that no broken or manipulated PATH can be passed via e.g. "su" or "sudo -E". This means that overrides must be placed in /usr/local/(s)bin now, which as well was the only save location for system-wide overrides before. Many thanks to @tandy-1000 for reporting a related issue: https://github.com/MichaIng/DietPi/issues/3873
- DietPi-Set_swapfile | Hardened permissions of the /dev/zram0 zram-swap device to be only accessible by root user (mode 0600) and interpret input argument "/dev/zram0" and "zram0" as zram-swap as well, aside of "zram" only. Many thanks to @hansjuergenmay for reporting a related issue: https://github.com/MichaIng/DietPi/issues/3869
- [DietPi-NordVPN](https://dietpi.com/docs/software/vpn/#dietpi-nordvpn) | A function has been added update the NordVPN server list, e.g. when new servers are available that are closer to your location.
- [DietPi-JustBoom](https://dietpi.com/docs/dietpi_tools/#configuration) | The ALSA equalizer settings are not stored to /var/lib/dietpi/dietpi-config/.alsaequal.bin where all users/services can read it and all members of the "dietpi" group (or root) can create and write to it. Many thanks to @wowcut for reporting this issue: https://github.com/MichaIng/DietPi/issues/3950
- DietPi-Update | It is now possible to enable automated APT package update checks and even upgrade them on a daily bases. Set the new dietpi.txt setting "CONFIG_CHECK_APT_UPDATES" to "1" (default) to enable daily APT update checks. The result is shown in the DietPi login banner, in a similar way an available DietPi update is presented. Set "CONFIG_CHECK_APT_UPDATES" to "2" to have APT packages upgraded automatically on a daily bases and review the logs at /var/tmp/dietpi/logs/dietpi-update_apt.log. Cron is used for the schedule, which means that you can control the execution time via dietpi-cron > "cron.daily". Many thanks to @ravenclaw900 for implementing this feature: https://github.com/MichaIng/DietPi/pull/3899
- DietPi-Config | RPi: When applying less than 32 MiB GPU memory, the VCSM (VideoCore Shared Memory) device and kernel module are disabled now. The automatically loaded cut-down firmware files do not support it which leads to currently 5 expected kernel error messages. Disabling it in the first place might marginally speed up boot and reduce memory usage, but more importantly reduces confusion or worries induced by to the error messages.
- DietPi-Config | Audio Options: Added an option to switch between direct audio output and automatic software conversions, for channels, format and rate via ALSA "plug" plugin. This may be required to play any raw wav file regardless of sound card capabilities and without defining supported values manually. Having automated software conversions enabled matches the Debian ALSA defaults but may increase CPU usage and decrease sound quality.
- [DietPi-Software | OpenBazaar](https://dietpi.com/docs/software/social/#openbazaar): Build is now done with the currently latest Go v1.15.3 and the service runs as unprivileged user "openbazaar" instead of root.
- [DietPi-Software | XRDP](https://dietpi.com/docs/software/remote_desktop/#xrdp): Remote desktop connections can now be done with the "Xorg" method and hence don't require an active VNC server anymore. New installes will not pull TigerVNC as dependency and if only RDP is required, TigerVNC can be uninstalled.
- [DietPi-Software | Single File PHP Gallery](https://dietpi.com/docs/software/social/#image-gallery): Updated new installs to latest v4.7.1. Run "dietpi-software reinstall 56" to upgrade your existing instance.
- [DietPi-Software | MineOS](https://dietpi.com/docs/software/gaming/#mineos): A systemd service is now used to run the daemon in favour of the previous supervisor, the obsolete "mineos" user is not created anymore, Node.js v11 is used to build MineOS, obsolete dependencies have been removed and a reinstall will now perform a MineOS upgrade and the existing /etc/mineos.conf is not replaced anymore.
- [DietPi-Software | Mycroft AI](https://dietpi.com/docs/software/hardware_projects/#mycroft-ai): By default, the ALSA automatic conversions plugin is enabled now on install, which enables OOTB compatibility with all wav sound formats and sound card. Additionally listener sampling rate override (48000 Hz) has been removed, reverting to Mycroft default (16000 Hz), which might solve an issue with hanging microphone level in CLI and related failing voice detection.
- [DietPi-Software | Gogs](https://dietpi.com/docs/software/cloud/#gogs): Has been enabled for ARMv8 boards and on ARMv7 boards the newest version is installed again, which was not the case since Gogs v0.12. due to changed download names. Many thanks to `@MadKernel` for bringing ARMv8 support back to our attention: https://dietpi.com/phpbb/viewtopic.php?t=8276
- [DietPi-Software | Folding@Home](https://dietpi.com/docs/software/computational_science/#foldinghome): Is now available for ARMv8 (arm64) boards. Donate your idle CPU power to help researchers fighting against diseases like Cancer, Alzheimer, Ebola and COVID-19.
- [DietPi-Software | Radarr](https://dietpi.com/docs/software/bittorrent/#radarr): Has been upgraded to the newest major version 3, including the required branch migration (from "develop" to "master") and, for non-ARMv6 boards, a migration to the now available .NET-based build away from Mono. This change is applied to all DietPi systems via Radarr reinstall, preserving your data and configs of course.
- [DietPi-Software | OctoPrint](https://dietpi.com/docs/software/printing/#octoprint): Is now available via PyPI repository, hence can be installed and updated easier and faster without downloading and building from the sources. Git is not required anymore, as the internal updater as well uses Python pip only. We migrate all DietPi OctoPrint instances to the new version and new install method, so that the Git/install directory /opt/octoprint is obsolete and removed. All your data will be preserved, of course. Many thanks to @Stinocon for reporting issues related to our previous implementation: https://github.com/MichaIng/DietPi/issues/3940

#### New Software Packages
- DietPi-Software | Portainer: Simplify your Docker container management via Portainer web interface, now available for install. Many thanks to @Joulinar for implementing this software option: https://github.com/MichaIng/DietPi/pull/3933
- DietPi-Software | Tor Relay: Become part of the Tor Project and turn your DietPi into a Tor relay to help others surf the Internet safely and privately. Many thanks to @ravenclaw900 for implementing this software option: https://github.com/MichaIng/DietPi/pull/3921
- [DietPi-Software | PaperMC](https://dietpi.com/docs/software/gaming/#papermc): Highly optimised Minecraft server, written in Java, extensible via plugins, now available for install. Many thanks to @ravenclaw900 for implementing this software option: https://github.com/MichaIng/DietPi/pull/3828
- [DietPi-Software | Unbound](https://dietpi.com/docs/software/dns_servers/#unbound): Validating, recursive, caching DNS resolver, now available for install and integrate with Pi-hole. Many thanks to @ravenclaw900 for implementing this software option: https://github.com/MichaIng/DietPi/pull/3872
- [DietPi-Software | Bitwarden_RS](https://dietpi.com/docs/software/cloud/#bitwarden_rs): An unofficial Bitwarden password manager server with web UI, written in Rust, now available for install. Many thanks to @CactiChameleon9 for implementing this software option: https://github.com/MichaIng/DietPi/issues/3724

#### Removed Software Packages
- DietPi-Software | CloudPrint: This software option has been removed, since the Google Cloud Print service will be shut down at the end of 2020 and we don't want to offer software options which will work for at most two months. Please migrate to another printing solution in time. Already installed CloudPrint instances will remain installed and the system service remains functional until the end of the year. With the first DietPi release in 2021 we will remove service handling and offer the package removal during the update process. Further information can be found here: https://www.google.com/cloudprint/learn/
- DietPi-Software | Tonido: This software option has been removed, since it is not developed anymore fore three years, no sources have been found and the latest binaries require ancient library versions which fail to be easily installed on currently supported Debian versions.

#### Bug Fixes
- DietPi-Config | Resolved an issue on RPi where the onboard 3.5mm jack selection did not work if an HDMI screen was attached. When selecting explicit either HDMI audio or onboard 3.5mm jack, the other one is now disabled on device tree level, which means it cannot be switched without reboot. This is currently the only known way to assure that 3.5mm jack is used for audio output regardless of now or later attached or detached HDMI screens. Many thanks to @corasaniti for reporting this issue: https://github.com/MichaIng/DietPi/issues/3887
- DietPi-Config | Resolved an issue on RPi where selecting the waveshare32 LCD panel installed an outdated device tree overlay, incompatible with the current Linux 5.4 kernel. Many thanks to @black00019 for reporting this issue: https://github.com/MichaIng/DietPi/issues/3881
- [DietPi-Services](https://dietpi.com/docs/dietpi_tools/#configuration) | Resolved an issue where CPU affinity selection failed due to a syntax error.
- [DietPi-Bugreport](https://dietpi.com/docs/dietpi_tools/#miscellaneous) | Resolved an issue where bug report uploads were cancelled if connection test on port 80/443 failed even that uploads are done via SFTP on port 22.
- [DietPi-Cleaner](https://dietpi.com/docs/dietpi_tools/#maintenance) | Resolved an issue where cleaner selection failed with a syntax error.
- [DietPi-Sync](https://dietpi.com/docs/dietpi_tools/#configuration) | Resolved an issue where setting the delete mode to "On" did not have any effect on it. Many thanks to `@mafioso12dk` for reporting this issue: https://dietpi.com/phpbb/viewtopic.php?t=8226
- DietPi-Banner + [DietPi-NordVPN](https://dietpi.com/docs/software/vpn/#dietpi-nordvpn) | Resolved an issue where WAN IP changed to a multi-line output. Many thanks to @maartenlangeveld for reporting this issue: https://github.com/MichaIng/DietPi/issues/3908
- [DietPi-CloudShell](https://dietpi.com/docs/software/system_stats/#dietpi-cloudshell) | Resolved an issue where scene selection failed with a syntax error. Many thanks to @ravenclaw900 for reporting this issue: https://github.com/MichaIng/DietPi/issues/3880
- [DietPi-Software | PiVPN](https://dietpi.com/docs/software/vpn/#pivpn): Resolved an issue where the installer hang since the interactive whiptail dialogues were not shown on console. Many thanks to @kelliegator for reporting this issue: https://github.com/MichaIng/DietPi/issues/3844
- [DietPi-Software | Medusa](https://dietpi.com/docs/software/bittorrent/#medusa): Resolved an issue where Medusa failed to start after install. Many thanks to @Luan7805 for reporting this issue: https://github.com/MichaIng/DietPi/issues/3842
- DietPi-Software | Webservers: Resolved an issue where reinstall failed if /var/www/html did not exist.
- DietPi-Software | Lighttpd: Resolved an issue where (re)install failed if the fastcgi or fastcgi-php module was enabled already.
- [DietPi-Software | O!MPD](https://dietpi.com/docs/software/media/#ompd): Resolved an issue where the URL check for youtube-dl failed.
- [DietPi-Software | Single File PHP Gallery](https://dietpi.com/docs/software/social/#image-gallery): Resolved an issue where directory previews were not shown due to missing permissions. Many thanks to `@tallbastard` for reporting this issue: https://dietpi.com/phpbb/viewtopic.php?p=28155#p28155
- [DietPi-Software | WebIOPi](https://dietpi.com/docs/software/hardware_projects/#webiopi): Resolved an issue where the download and install failed.
- DietPi-Software | Nginx: Resolved an issue where the amount of worker processes was not set to the amount of CPU threads as intended.
- [DietPi-Software | MineOS](https://dietpi.com/docs/software/gaming/#mineos): Resolved an issue where the Node.js downgrade an hence the web UI compiling failed. Many thanks to @CactiChameleon9 for reporting this issue: https://github.com/MichaIng/DietPi/issues/3901
- [DietPi-Software | Mycroft AI](https://dietpi.com/docs/software/hardware_projects/#mycroft-ai): Resolved an issue where skill installs failed due to missing permissions. Many thanks to `@GreenGentleman` for reporting this issue: https://dietpi.com/phpbb/viewtopic.php?p=28464#p28464
- DietPi-Software | Jellyfin: Resolved an issue where Jellyfin did not run on ARMv6 (RPi 1 and Zero models). ARMv6/Raspbian-compatible binaries are now shipped via different APT repository. Many thanks to `@Shiivu` for reporting this issue: https://dietpi.com/phpbb/viewtopic.php?t=8263
- DietPi-Software | WiFi Hotspot: Resolved an issue on RPi with WiFi adapters using an RTL8188C* chip, where the service failed to start due to an invalid driver. Many thanks to `@yerc` for reporting this issue: https://dietpi.com/phpbb/viewtopic.php?p=28984#p28984
- [DietPi-Software | OctoPrint](https://dietpi.com/docs/software/printing/#octoprint): Resolved an issue where the service and system could not be restarted anymore via OctoPrint web UI and, on RPi, the CPU throttling state could not be derived.
- [DietPi-Software | Ampache](https://dietpi.com/docs/software/media/#ampache): Resolved an issue where the install failed due to an incompatible PHP composer version. Many thanks to `@kk345` for reporting this issue: https://dietpi.com/phpbb/viewtopic.php?t=8367

MichaIng committed:
Release v6.32 (#3751)

MichaIng committed:
v6.33

+ META | Moving some images over to the website repo

MichaIng committed:
v6.33

+ META | Moving some images over to the website repo

MichaIng committed:
Release v6.33 (#3829)

MichaIng committed:
v6.34 (#3836)

+ META | Merge "merge commits" from master back to dev

MichaIng committed:
v6.34

+ DietPi-Globals | Default subversion up

MichaIng committed:
v6.34

+ Subversion up

MichaIng committed:
v6.34

+ DietPi-Patch | Init v6.34 patches

Joulinar committed:
DietPi-Update | update medusa service

medusa service needs to be updated as user group medusa is not existing

MichaIng committed:
v6.34

+ DietPi-Software | Medusa: Be specific/failsafe when commenting "Group=" in systemd unit

MichaIng committed:
v6.34 (#3845)

+ DietPi-Software | Show output of PiVPN installer

MichaIng committed:
v6.34 (#3846)

+ Update "medusa" branch with CHANGELOG.txt v6.34 init

MichaIng committed:
v6.34

+ CHANGELOG | Medusa: Resolved an issue where Medusa failed to start after install

MichaIng committed:
v6.34 (#3843)

+ DietPi-Software | Medusa: Fix failing removal of group from systemd unit

MichaIng committed:
v6.34 (#3855)

+ DietPi-Bugreport | Do not check if upload address is reachable via HTTP/HTTPS but check on SFTP/SSH port explicitly since this is how the bug report is uploaded
+ DietPi-Bugreport | Massively simplify code by using DietPi-Globals

MichaIng committed:
v6.34

+ DietPi-Imager | Fix and add menu pre-selections

MichaIng committed:
v6.34 (#3848)

+ DietPi-Software | OpenBazaar: Update Go to current v1.15.3
+ DietPi-Software | OpenBazaar: Run service with unprivileged user "openbazaar"
+ DietPi-Software | OpenBazaar: Disable file logging "-f" which forces logs to STDOUT (=systemd-journald); Reduce log level to "notice" to avoid verbose debug logging to system journal
+ DietPi-Software | OpenBazaar: Update it on reinstall i.e. when it has been installed already through go get
+ DietPi-Software | OpenBazaar: On uninstall, remove binary and archive via "go clean"

MichaIng committed:
v6.34 (#3860)

+ DietPi-Software | CloudPrint: Remove from software options
+ DietPi-Patch | CloudPrint: Do not reinstall it anymore, since this install option has been removed
+ DietPi-Patch | CloudPrint: Unset CloudPrint install flag to free up ID
+ DietPi-Survey_report | Add Bazarr with v6.33
+ DietPi-Survey_report | Add v6.34 support and remove CloudPrint with this version

MichaIng committed:
v6.34

+ DietPi-Obtain_HW_model | Fix RPi CM4 detection, the revision code ID is 14, not 12: https://github.com/raspberrypi/documentation/pull/1616

nonnorm committed:
v6.34 (#3854)

+ DietPi-Software | Add Create_Config function which starts a service and waits for a config file to be pre-created and optionally content to be present. This is useful when we need the current default config file present on a fresh install to alter it.

MichaIng committed:
v6.34 (#3865)

+ DietPi-Software | Lighttpd: lighty-enable-mod "fails" with exit code 2 if a module is enabled already. Override this exit code with success (0) when using G_EXEC.
+ DietPi-Software | Consequently use G_EXEC for all lighty-enable-mod calls and override exit code 2 to run cause the same bug anymore.
+ DietPi-Software | Webservers: Do not attempt to remove /var/www/html if it does not exist

nonnorm committed:
v6.34 (#3828)

+ DietPi-Software | PaperMC: Add install option as ID 181
+ DietPi-Software | Create_Config: Adjust default timeout to mostly used 25 seconds and print how long it actually took
+ DietPi-Software | Create_Config: Minor output enhancement
+ DietPi-Software | PaperMC: Install mcrcon to /usr/local/bin/ directly, in theory it can be used with any Minecraft server (?) and is not used by the PaperMC service itself anymore.

MichaIng committed:
v6.34 (#3868)

+ CHANGELOG | PaperMC: Highly optimised Minecraft server, written in Java, extensible via plugins, now available for install
+ DietPi-Survey_report | Add PaperMC and fix a few titles capitalisation

MichaIng committed:
v6.34 (#3870)

+ DietPi-Set_swapfile | Harden zram-swap device permissions to be 0600
+ DietPi-Set_swapfile | Be more tolerant to interpret "zram0" and "/dev/zram0" as zram-swap as well

MichaIng committed:
v6.34

+ DietPi-Globals | G_INIT: Set PATH in scripts to system default, to rule out missing commands due to broken environment, e.g. due to "su" or "sudo -E"

MichaIng committed:
v6.34

+ CHANGELOG | DietPi-Globals: In DietPi scripts, the PATH variable is now overwritten with the Debian/bash system default to assure that no broken or manipulated PATH can be passed via e.g. "su" or "sudo -E".

MichaIng committed:
v6.34

+ DietPi-Banner | Syntax fix

MichaIng committed:
v6.34

+ CHANGELOG | DietPi-CloudShell: Resolved an issue where screne selection failed with a syntax error

MichaIng committed:
v6.34

+ DietPi-Cleaner | Syntax fix

MichaIng committed:
v6.34

+ DietPi-Services | Syntax

MichaIng committed:
v6.34

+ CHANGELOG | Add DietPi-Services and DietPi-Cleaners syntax fix

MichaIng committed:
v6.34

+ DietPi-Set_software | RPi waveshare32 LCD panel: Install latest Linux 5.4 compatible device tree overlay directly from waveshare GitHub repository
+ DietPi-Set_software | Apply G_EXEC error handler on all commands with no stream redirects

MichaIng committed:
v6.32

+ CHANGELOG | DietPi-Config: Resolved an issue on RPi where selecting the waveshare32 LCD panel installed an outdated device tree overlay, incompatible with the current Linux 5.4 kernel. Many thanks to @black00019 for reporting this issue: https://github.com/MichaIng/DietPi/issues/3881

MichaIng committed:
v6.34

+ DietPi-Sync  | Fix v6.33 regression: Overseen usage of old variable "rsync_options" which broke delete mode selection

MichaIng committed:
v6.34

+ CHANGELOG | DietPi-Sync: Resolved an issue where setting the delete mode to "On" did not have any effect on it.

MichaIng committed:
v6.34

+ DietPi-Software | WireGuard: Fix/extend WireGuard module detection due to changed path inside of the kernel module directory structure

MichaIng committed:
v6.34 (#3888)

+ DietPi-Software | XRDP: Install xorgxrdp package to allow RDP connections via "Xorg" method without VNC and do not install TigerVNC together with XRDP anymore
+ DietPi-Patch | XRDP: Install xorgxrdp, if missing, to allow RDP connections via "Xorg" method without active VNC server

Joulinar committed:
v6.34 (#3879)

+ DietPi-NordVPN | Add a function to update NodVPN server list
+ DietPi-NordVPN | Apply new coding standard
+ DietPi-Patch | Syntax

Joulinar committed:
v6.34 (#3889)

+ DietPi-Software | Fail2Ban: Fix drop-in config directory pre-creation

nonnorm committed:
v6.34 (#3872)

+ DietPi-Software | Add "Unbound" recursive DNS server

Joulinar committed:
v6.34 (#3890)

+ DietPi-Software | O!MPD: Skip G_CHECK_URL for youtube-dl download: Due to DCMA violations, it has been taken down from GitHub, but the direct download link has been changed to remain functional. We'll keep an eye on the legal state of youtube-dl in general, but for now, as it allows access to legal and illegal content like any other download tool or browser does, we see no reason to drop support our side. Since we are no specialists, we follow bigger Linux distributions in this regards, and the Debian repository still contains and allows to install youtube-dl, so do we.
+ DietPi-Software | O!MPD: Handle extracted OMPD-master dir explicitly and use G_EXEC where applicable

MichaIng committed:
v6.34

+ DietPi-Software | Fix changed DietPi logo URL

MichaIng committed:
v6.34 (#3896)

+ DietPi-Software | Single File PHP Gallery: Fix permissions for cache directory
+ DietPi-Software | Assure that required PHP gd module is installed and enabled
+ DietPi-Software | Update to v4.7.1 and provide download via 7z file without version string, which allows us to update to version outside of DietPi updates.

MichaIng committed:
v6.34

+ DietPi-Software | Xserver: Fix removal of DPMS config on uninstall

MichaIng committed:
v6.34

+ DietPi-Set_swapfile | Re-assure that any swap file is disabled before attempting dd allocation fallback, since the previous swapon (after fallocate allocation) might have succeeded partially, producing an error exit code but still leading to a busy swap file, which then leads to failing dd.

MichaIng committed:
v6.34

+ DietPi-Software | Download_Install: Resolve failing downloads when the URL does not contain a file name with extension and the resulting file type could hence contain a slash, being interpreted as relative directory path. This broke WebIOPi download and extraction and would do so for any other SourceForge download where the URL looks like ..."/something.tar.gz/download".

MichaIng committed:
v6.34

+ CHANGELOG | WebIOPi: Resolved an issue where the download and install failed.

Joulinar committed:
v6.34 (#3902)

+ DietPi-Software | Nginx: correct typo on configuration section

MichaIng committed:
v6.34 (#3900)

+ DietPi-Config | RPi: Disable VCSM if less than 32 MiB GPU memory are applied, since in this case the cut-down firmware is loaded which does not provide the features required for VCSM to succeed. This may reduce boot overhead and memory usage marginally but especially mutes 5 related kernel error messages at boot.
+ DietPi-Patch | RPi: Re-apply GPU memory split to disable VCSM if GPU memory is lower than 32 MiB
+ DietPi-PREP | RPi: Apply GPU memory split to have the dtoverlay created which mutes VCSM related error messages when less than 32 MiB GPU memory are applied

MichaIng committed:
v6.34 (#3903)

+ DietPi-Software | Tonido: Drop support as this project has been obviously abandoned (no sign since 2017), no source code available (if it ever was) and the pre-compiled binaries depend on ancient libpng12 which is not available anymore since Debian Stretch (=libpng16) and now installing the package from Jessie fails with a strange error, most likely due to a change in apt or dpkg.
+ DietPi-Patch | Remove Tonido install flag and obsolete reinstalls
+ DietPi-Services | Remove Tonido from DietPi control
+ DietPi-Survey_report | Remove Tonido with v6.34
+ DietPi-Patch | Tonido: Preserve DietPi-Services control if it has not been excluded as it has been removed from the static list

MichaIng committed:
v6.34

+ DietPi-Software | Align category and software names with DietPi-Docs

MichaIng committed:
v6.34 (#3911)

+ DietPi-Banner + DietPi-NordVPN: Resolved an issue where WAN IP changed to a multi-line output

MichaIng committed:
DietPi-Software | MineOS: Several updates (#3904)

+ DietPi-Software | MineOS: Fix Node 8 install and remove obsolete npm and "n" options
+ DietPi-Software | MineOS: Use latest compatible Node v11
+ DietPi-Software | MineOS: Switch form supervisor to systemd unit
+ DietPi-Software | MineOS: Remove Python 3 from dependencies, as this was required for the Python-based implementation only, while this is a Node.js-based one.
+ DietPi-Software | MineOS: Do not create obsolete "mineos" user anymore
+ DietPi-Patch | MineOS: Reinstall MineOS to migrate from supervisor to systemd unit, if it is not yet present

MichaIng committed:
v6.34

+ DietPi-Config | Re-allow adjusting arm_freq_min on all RPi models, it is mostly stable, only requires a non-lowered over_voltage, which is trial&error value anyway.

MichaIng committed:
v6.34

+ DietPi-Patch | Do not comment arm_freq_min anymore, which is mostly stable with current kernel release

MichaIng committed:
v6.34 (#3916)

+ DietPi-Software | Mycroft AI: Pre-create the skills directory since the installer chowns our parent symlink instead of the directory itself, which leads to Mycroft AI not being able to install skills.
+ DietPi-Software | Mycroft AI: Remove logs dir and sudoers file on uninstall
+ DietPi-Software | Mycroft AI: Apply G_EXEC for command transparency and error handling where possible
+ DietPi-Set_hardware | Soundcard: Allow applying the "plug" ALSA plugin for automated channels, rate and format conversion, required to play wav files with incompatible values, e.g. required for Mycroft AI to support all voices. This plugin is also used by default on Debian when no asound.conf override exists.
+ DietPi-Config | Allow switching the ALSA automatic conversion plugin (plug)
+ DietPi-JustBoom | Fix equalizer toggle by removing the suffix when reading the soundcard from dietpi.txt, and remove the new -plug suffix for auto-conversion plugin as well.
+ DietPi-JustBoom | Re-read sound card index on every menu loop, to be failsafe, e.g. if the sound card is changed concurrently by another script/session
+ DietPi-Software | Mycroft AI: Enable ALSA automatic software conversion plugin, which is often required to play Mimic-generated wav files.
+ DietPi-Software | Mycroft AI: Remove PulseAudio from installer dependencies, we use ALSA
+ DietPi-Software | Mycroft AI: Do not force listener sample rate anymore. With automatic software conversions (ALSA "plug" plugin) all works and in some cases, for unknown reason, with 48000 Mycroft lost microphone connection, i.e. CLI mic bar freezes => no voice detection.

MichaIng committed:
v6.34

+ DietPi-Software | XRDP: Apply workaround for missing mouse and keyboard input, which is not always but sometimes still required.

MichaIng committed:
v6.34

+ DietPi-Patch | XRDP: Apply workaround for for failing mouse and keyboard input, which not not always but sometimes still required.

nonnorm committed:
v6.34 (#3899)

+ DietPi-Update | If no DietPi update has been found, or when invoking via "dietpi-update 2", run optional (1) APT update checks or (2) automated APT upgrades.
+ dietpi.txt | Add a new setting to enable automated daily APT update checks (=1, default) or even upgrades (=2), invoked by the daily DietPi cron job.
+ DietPi-Software | PiVPN: Since the installer asks to enable "unattended-upgrades" which is masked by default on DietPi, override that choice from the installer but instead offer to enable DietPi daily APT upgrades instead.
+ DietPi-Banner | Show an info if available APT upgrades have been found during daily check or manual "dietpi-update 2" invocation.
+ General | Some code enhancements

MichaIng committed:
v6.34

+ DietPi-Software | Jellyfin: Fix implementation for ARMv6 as this is not served via dedicated Raspbian-compatible repository

MichaIng committed:
v6.34

+ CHANGELOG | Jellyfin: Resolved an issue where Jellyfin did not run on ARMv6 (RPi 1 and Zero models). ARMv6/Raspbian-compatible binaries are now shipped via different APT repository. Many thanks to @Shiivu for reporting this issue: https://dietpi.com/phpbb/viewtopic.php?t=8263
+ CHANGELOG | Shorten all phpBB forum URLs

MichaIng committed:
v6.34 (#3922)

+ DietPi-Software | Gogs: Fix Git archive file names and enable for ARMv8
+ DietPi-Software | Gogs: Use G_EXEC where applicable for error handling and to increase verbosity/transparency
+ DietPi-Software | Gogs: Update systemd unit with hardenings from upstream one
+ DietPi-Software | Gogs: Ignore errors on SQL import due to the attempt to set innodb_file_format and innodb_large_prefix
+ DietPi-Software | Gogs: Do no redirect service logs to a file anymore, as it by default only contains a single line, telling that file logging is used. Service failures of other kind anyway need to be reviewed via "journalctl -u gogs", so users should get used to check it aside of the log files.

MichaIng committed:
v6.34 (#3924)

+ DietPi-Survey_report | Add Unbound to list with v6.34
+ DietPi-Software | Unbound: Add online docs URL

MichaIng committed:
v6.34

+ DietPi-Autostart | Remove RetroPie autostart option until we find time retest and in case re-implement it

MichaIng committed:
Update dietpi-login

+ DietPi-Login | Remove RetroPie autostart option until we find time to retest and in case re-implement it

MichaIng committed:
v6.34

+ dietpi.txt | Remove obsolete RetroPie autostart option

MichaIng committed:
v6.34

+ DietPi-Globals | G_CHECK_URL: Switch from "wget --spider" to "curl -ILfv" to check URLs, which is often a bid faster, gives more detailed output, usable for debugging, and that way we consequently use curl for all kind of HTTP requests.

MichaIng committed:
v6.34

+ DietPi-Bugreport | Remove separate G_CHECK_URL before uploading, as curl correctly does not report success on port 22 and the upload itself will show the same error anyway, if our server is not reachable. To give some more details on G_EXEC error handling, add "-v" verbose flag. It seems strange in combination with "-s" silent, but that one especially mutes processing output and received document content.
+ DietPi-Bugreport | Increase upload connection timeout, as slow network might be even a reason for the bug report.

MichaIng committed:
v6.34

+ DietPi-Software | Minor

MichaIng committed:
v6.34

+ DietPi-Globals | G_CHECK_URL: Fix timeout due to move from wget to curl

MichaIng committed:
v6.34

+ DietPi-PREP | ROCK Pi S: Move to new (renamed) kernel package and skip the obsolete u-boot meta package. As time of writing, due to a broken apt,radxa.com repo (expired key + broken dependencies) this fails, but will likely be the right setup once the issues have been resolved.

MichaIng committed:
v6.34

+ DietPi-Imager | Fix backup GPT table after the final image file has been created, since the table is created at the end of the device, which is usually truncated.

MichaIng committed:
v6.34

+ DietPi-Config | RPi GPU memory split: Do not re-compile the device tree overlay on every kernel update, which is not required in this simple case and there is no practical change that even on fundamental kernel changes that overlay breaks anything with the very clear device path it disables. Remove the forgotten module blacklist when setting >= 32 MiB GPU memory.

MichaIng committed:
v6.34

+ DietPi-Patch | Update ASIX kernel driver only on RPi CM 3+ (=USBridgeSig) and only for v7+ kernel

MichaIng committed:
v6.34

+ DietPi-Patch | Update ASIX kernel driver only on RPi CM 3+ (=USBridgeSig) and only for v7+ kernel, but force once (overriding check) during DietPi-PREP to have a USBridgeSig ready image.

MichaIng committed:
v6.34

+ DietPi-Software | Home Assistant: Error handle pyenv install

MichaIng committed:
v6.34

+ DietPi-PREP | Cleanup and enhance Radxa ROCK Pi S image by applying the Radxa repo with updated key (the one from the current image expired at 22.10.), using HTTPS and removing some obsolete enabled service links which are not removed together with their DEB packages. For such links is now search for all services known to be left on pre-images but unwanted or obsolete.

MichaIng committed:
v6.34

+ DietPi-Software | WiFi Hotspot: Fix wrong hostapd driver selection in case of RTL8188C* chip: Stay with "nl80211" on RPi where the originating bug has been solved differently, apply "rtl871xdrv" only if our own compatible binary or a "hostapt-realtek" DEB package has been installed before.

MichaIng committed:
v6.34

+ CHANGELOG | WiFi Hotspot: Resolved an issue on RPi with WiFi adapters using an RTL8188C* chip, where the service failed to start due to an invalid driver

CactiChameleon9 committed:
v6.34 (#3894)

+ DietPi-Software | Adding Bitwarden_RS, an unofficial Bitwarden password manager server with web UI, written in Rust

MichaIng committed:
v6.34 (#3934)

+ CHANGELOG | Bitwarden_RS: An unofficial Bitwarden password manager server with web UI, written in Rust, now available for install
+ README | Add newly added software source code links
+ DietPi-Survey_report | Add support for Bitwarden_RS and apply some software name spelling fixes

MichaIng committed:
v6.34

+ DietPi-Banner | Syntax

MichaIng committed:
v6.34 (#3935)

+ DietPi-Software | Folding@Home: Enable install on ARMv8, which is possible with the latest version

MichaIng committed:
v6.34

+ DietPi-Optimal_MTU | ping via IPv4 explicitly since we assume ICMPv4 headers

MichaIng committed:
v6.34

+ DietPi-Login | Move aliases from DietPi-Globals to interactive bash script, since aliases have not meaning in functions

MichaIng committed:
v6.34

+ DietPi-Globals | Aliases have been moved from DietPi-Globals to interactive bash script, since those have no meaning in functions

MichaIng committed:
v6.34

+ DietPi-Pre-patches | Fix Radxa APT repository on ROCK Pi S

MichaIng committed:
v6.34

+ DietPi-Patch | Repair Radxa ROCK Pi S repository and marked packages

MichaIng committed:
v6.34

+ DietPi-Obtain_HW_model | Detect Raspberry Pi 400 correctly: https://github.com/raspberrypi/documentation/commit/ff3b562d6044cb50545104a04567614feb09dd91

MichaIng committed:
v6.34

+ CHANGELOG | Correct detection for the new Raspberry Pi 400 has been added

MichaIng committed:
v6.34

+ CHANGELOG | Remove known issues that we do not track any longer

MichaIng committed:
v6.34

+ DietPi-Obtain_HW_model | RPi 400 has onboard WiFi

MichaIng committed:
v6.34

+ DietPi-FirstBoot | RPi 400 ARM by default runs at 1800 MHz

MichaIng committed:
v6.34

+ CHANGELOG | Shorten phpBB query strings (forum ID is obsolete if topic or post ID is given, topic ID is obsolete if post ID is given)

MichaIng committed:
v6.34

+ README | Shorten phpBB query string

MichaIng committed:
v6.34

+ DietPi-Patch | Shorten phpBB query strings

MichaIng committed:
v6.34

+ DietPi-PREP | Shorten phpBB query strings

MichaIng committed:
v6.34

+ DietPi-Drive_Manager | Shorten phpBB query string

MichaIng committed:
v6.34

+ DietPi-CloudShell | Shorten phpBB query string

MichaIng committed:
v6.34

+ DietPi-Patch | Sparky SBC: Update Ethernet driver: https://github.com/sparky-sbc/sparky-test/tree/master/sparky-eth

MichaIng committed:
v6.34

+ DietPi-Globals | Fix hierarchy system: The first level script must start with "0" instead of "1"

MichaIng committed:
v6.34 (#3945)

+ DietPi-Software | Download_Install: Fix fallback URL usage
+ DietPi-Software | Radarr: Migrate to .NET core version on non-ARMv6 devices
+ DietPi-Software | Radarr: Move install dir to /opt/radarr (lower case characters)
+ DietPi-Software | Jackett: Fix .NET dependency for Bullseye: libicu67
+ DietPi-Patch | Reinstall Radarr to update to newest version, migrate to .NET based build on non-ARMv6 boards and apply new install dir /opt/radarr (lower case)
+ DietPi-Patch | Radarr: Apply branch change to master as being the replacement since v3

MichaIng committed:
v6.34

+ DietPi-PREP | Overwrite keyboard layout already during DietPi-PREP to prevent the case of failure due to invalid entries. It is overwritten on first boot by dietpi.txt setting anyway, so no point to preserve previous choices here.

MichaIng committed:
v6.34 (#3949)

+ DietPi-Software | OctoPrint: Switch to now available PyPI repo install, which matches to what the internal updater does. For now we'll stay with a system-wide install, but this required some workaround to permit and force being used as well by OctoPrint for all its pip actions.
+ DietPi-Software | OctoPrint: Grant permissions to run vcgencmd (for CPU throttling information) on RPi via "video" group
+ DietPi-Software | OctoPrint: Pre-apply the service/system restart/shutdown commands, as those require sudo now
+ DietPi-Patch | OctoPrint: Reinstall to migrate to new PyPI-based install and apply permission fixes
+ DietPi-Software | OctoPrint: Fix pip uninstall and do not remove other users .octoprint data anymore as this has been migrated and we might remove data from a custom OctoPrint instance that is not wanted.
+ DietPi-Software | OctoPrint: The PyPI-based install does not take that long since many modules are pre-compiled wheels, hence remove the obsolete info print which doubles a bid with G_EXEC info anyway.

nonnorm committed:
v6.34 (#3921)

+ DietPi-Software | Tor Relay: Now available for install. It is currently incompatible with Tor Hotspot, hence assure that both are never installed together.
+ DietPi-Software | Squeezelite: Renamed from "SqueezeLite" to "Squeezelite"
+ DietPi-CloudShell | Tor Relay monitor scene added
+ DietPi-CloudShell | Reset colour after intro animation

nonnorm committed:
v6.34 (#3951)

+ DietPi-Software | Move WiFi Hotspot, Tor Hotspot and No-IP to Advanced Networking category and remove WiFi Hotspot category

MichaIng committed:
v6.34

+ DietPi-Software | Webmin: Remove obsolete workaround

MichaIng committed:
v6.34

+ DietPi-Patch | Webmin: Remove obsolete workaround: https://github.com/MichaIng/DietPi/issues/3810

Joulinar committed:
v6.34 (#3933)

+ DietPi-Software | Portainer: Add new software title
+ DietPi-Software | Docker: Remove config directory during uninstall
+ DietPi-Software | Fix alignment, trailing spaces and doubled empty lines

MichaIng committed:
v6.34 (#3953)

+ DietPi-JustBoom | Move ALSA equalizer config to world-readable location
+ DietPi-Set_hardware | Move ALSA equalizer config to world-readable location
+ DietPi-Config | The ALSA equalizer plugin implies the auto-conversion plugin, hence when equalizer is enabled, show auto-conversion as enabled, too.
+ DietPi-Patch | Repair sound card setting, in case "-eq" got appended multiple times: https://github.com/MichaIng/DietPi/issues/3950
+ DietPi-Patch | Migrate ALSA equalizer config to new location

MichaIng committed:
v6.34

+ DietPi-Globals | G_DIETPI-NORIFY/G_CONFIG_INJECT: Remove colour codes from output if no real terminal is attached, to clean up log file/journal output

MichaIng committed:
v6.34 (#3938)

+ DietPi-Set_hardware | soundcard: Fix RPi onboard audio which cannot be switched via amixer controls anymore. 3.5mm jack and HDMI (on RPi4 each port) are separate sound cards now. This solution disables either 3.5mm jack or HDMI audio via device tree overlay, specially to assure that HDMI audio can not move the 3.5mm jack from card0 to card1. Disabling 3.5mm when HDMI audio is forced might save some resources, which it would always be card1, hence never affects the index or the HDMI audio device.

MichaIng committed:
v6.34

+ DietPi-Update | Do not abort anymore core version mismatch. We'll always provide a migration branch that will be provided via new server version file or for older versions via (pre-)patches.
+ DietPi-Update | Do not apply APT upgrades on check only mode 2, which is e.g. called on boot as well.

MichaIng committed:
v6.34

+ DietPi-Cron_daily | Apply APT upgrades within cron script, in case chosen via dietpi.txt, since "dietpi-update 2" only applies the check now

MichaIng committed:
v6.34

+ DietPi-Update | Syntax

MichaIng committed:
v6.34

+ DietPi-Update | Syntax

MichaIng committed:
v6.34

+ DietPi-Globals | Assure index "i" is declared as local variable in all functions

MichaIng committed:
v6.34

+ README | Add missing/new software sources for: Docker, Portainer, Tor

MichaIng committed:
v6.34

+ CHANGELOG | Add Portainer and Tor Relay changelog entries

MichaIng committed:
v6.34

+ DietPi-Survey_report | Add Tor Relay and Portainer support

MichaIng committed:
v6.34

+ DietPi-Globals | G_AGUP allows now to store the number of available APT upgrades to either a file (-f [file path]) or a variable (-v [variable name]) with file path and variable name optional, defaulting to /run/dietpi/.apt_updates and G_AGUP_COUNT respectively.
+ DietPi-Globals | Assure that exit codes are not overwritten by "unset".

MichaIng committed:
v6.34

+ DietPi-Update | Use new G_AGUP CLI to check for number of available APT updates

MichaIng committed:
v6.34

+ DietPi-Cron_daily | Use new G_AGUP CLI to check for available number of APT upgrades

MichaIng committed:
v6.34

+ DietPi-Globals | G_AGUP: Mute "apt" CLI warning

MichaIng committed:
v6.34

+ DietPi-Globals | G_AGUP: Fix input detection

Joulinar committed:
v6.34  (#3956)

+ DietPi-Software | Ampache: Change Composer version to latest 1.x as Composer version 2.x is not compatible to Ampache

Joulinar commented:
uhhh quite a big release with a lot of new stuff

nonnorm commented:
Correct me if I'm wrong, but I don't see the APT update feature in the changelog.  It's a fairly large new feature.

MichaIng commented:
Good point, will add it.
... done: https://github.com/MichaIng/DietPi/commit/cbad6eb21fa2194b95e5fccd9189a2a4e676873e

I btw refined the feature today a bid. E.g. `dietpi-update 2` does not perform APT upgrades anymore, as this was meant to be a check-only input argument and is executed e.g. at the end of boot, to check for DietPi updates. Instead, only `dietpi-update 1` does APT upgrades (when enabled via dietpi.txt setting), if no DietPi update is available, and any way together with the DietPi update.

And to have a central CLI for this:
- `G_AGUP -f [optional file path]` stores the amount of available APT upgrades to the file, defaulting to `/run/dietpi/.apt_updates`.
- `G_AGUP -v [optional variable name]` stores it to a variable, defaulting to `G_AGUP_COUNT`.

I didn't add a documentation about this yet, as I want to play around with it and probably adjust.

MichaIng committed:
v6.34

+ CHANGELOG | DietPi-Update: It is now possible to enable automated APT package update checks and even upgrade them on a daily bases. Set the new dietpi.txt setting "CONFIG_CHECK_APT_UPDATES" to "1" (default) to enable daily APT update checks. The result is shown in the DietPi login banner, in a similar way an available DietPi update is presented. Set "CONFIG_CHECK_APT_UPDATES" to "2" to have APT packages upgraded automatically on a daily bases and review the logs at /var/tmp/dietpi/logs/dietpi-update_apt.log. Cron is used for the schedule, which means that you can control the execution time via dietpi-cron > "cron.daily". Many thanks to @ravenclaw900 for implementing this feature: https://github.com/MichaIng/DietPi/pull/3899

MichaIng committed:
v6.34

+ DietPi-Drive_Manager | btrfs-tools has been renamed to btrfs-progs with Debian Stretch. The first was a transitional dummy package, available until Debian Buster only.

StephanStS committed:
v6.34 (#3962)

CHANGELOG:
- "New Software Packages" grouped
- "Removed Software Packages" grouped
- Typo: "DietPi-Cleaners" => "DietPi-Cleaner"

MichaIng committed:
v6.34

+ RC up

MichaIng committed:
v6.34

+ DietPi-Globals | Default RC up

MichaIng committed:
v6.34

+ DietPi-Survey_report | Fix Single File PHP Gallery name
